### PR TITLE
Add Assessment Score UI

### PR DESCRIPTION
**Pull request by sderickson**
With open-ended levels having score and bronze/silver/gold thresholds, this PR adds UI to show this to students and teachers.
* Student Assessments View shows score and badge
* Teacher Assessments View assessments tab shows badge in page, full score and badge in popover.
* Tweaked the already-existing score in the Play Level View, moving the badge and adding a drop-shadow.

This also includes engineering work which should be reviewed.

# Sharing logic between Vue, Backbone
Instance methods Level.thresholdForScore and LevelSession.getTopScores were converted to static, and the instance methods became a thin wrapper around the static methods. This way pages that continue to be Backbone (like the LevelPlayView) and pages that use Vue (like the StudentAssessmentsView) can share that logic. An instance method tends to use Backbone models, the static methods use raw data, and so an instance method will typically convert the backbone models to raw JSON with `toJSON()`.

Questions:
* Are there better ways to share logic like this between Backbone and Vue pieces?
* How can this pattern's key characteristics and purposes best be communicated?

# Splitting Concerns Between Components
I refactored the StudentAssessmentRow component out of the StudentAssessmentsView. This is a good opportunity to leverage components to organize logic. Part of splitting this logic apart was deciding where pieces like parsing topScores out of Session and Level data. I decided to have all the data processing in the View and the presentation logic in the Row. So instead of

`student-assessment-row(:level=level, :session=session)`

I have

`student-assessment-row(:assessment=level.assessment, ...)`

This ends up being more verbose, but it is more compartmentalized, especially regarding the data and how it's fetched or projected. The Row just gets the raw data and presents it, while the View fetches the data and projects what it needs to give to the Row. The Row also clearly states what data it needs in order to do its job, in the list of props. This is in stark contrast with the TeacherClassView which needed a project string in the server-side Classroom model updated in order to show the data in the template, creating a distant dependency.

Questions:
* Does this seem like a good approach to breaking down larger components?

### CODE REVIEW

**Comment by differentmatt** [comment]
Yes, this seems like a fine way to break down components.

Couple questions:
Why didn't we convert parent student assessments view to single .vue file too?
Why did we choose CoffeeScript instead of JavaScript for new row component?

**Comment by differentmatt** [comment]
Why don't we update the legacy Backbone calls to use the new static methods too, and get rid of the instance methods completely?

**Comment by sderickson** [comment]
Good point, good opportunity to convert StudentAssessmentsView to a `vue` file. Will do!
Also good point about Backbone static methods. In cases where it's a lot to refactor all calls to the instance method, would defer, but in this case it would be pretty easy.
Need to set up an ES6 loader before we start migrating away from CS.

File: server/models/Classroom.coffee [hunk 1]
```diff
@@ -166,7 +166,7 @@ ClassroomSchema.methods.fetchSessionsForMembers = co.wrap (members) ->
       memberCoursesMap[userID.toHexString()] ?= []
       memberCoursesMap[userID.toHexString()].push(courseInstance.courseID)
   dbqs = []
-  select = 'state.complete level creator playtime changed created dateFirstCompleted submitted published code codeConcepts'
+  select = 'state.complete state.topScores level creator playtime changed created dateFirstCompleted submitted published code codeConcepts'
```
[the remark this answers is not in the document]
Comment (sderickson) [reply server/models/Classroom.coffee]: Refactor the TeacherClassView. In general for a smaller page, use the structure where the root node passes all info to child nodes, and for larger pages have all the state management in vuex.

**Pull request merged**